gcovr/configuration.py

```diff
@@ -641,6 +641,13 @@ def choose(_cls, choices, default=None):
         default=None,
         const=OutputOrDefault(None),
     ),
+    GcovrConfigOption(
+        "html_details_syntax_highlighting", ["--html-details-no-syntax-highlighting"],
+        group="output_options",
+        help="Disable syntax highlighting in HTML details page. "
+             "Default is active syntax highlighting.",
+        action="store_false",
+    ),
```

Comment:
In the future, it's worth considering if we want to name negative options as `--NAMESPACE-no-FEATURE` or `--no-NAMESPACE-FEATURE`. I'm in the second camp because this allows positive/negative pairs to be autogenerated. 

For example:

```python
GcovrConfigOption(
  "html_details_syntax_highlighting", ["--html-details-syntax-highlighting"],
  group="output_options",
  help="Use syntax highlighting in HTML details page. "
       "Enabled by default.",
  action="store_const",
  default=True,
  const=True,
  const_negate=False,  # autogenerates --no-NAME with action const=False
)
```

Until now, the only use of this feature is the `--html-self-contained`/`--no-html-self-contained` option.

There are also concerns regarding configuration file support. A config key is derived for all options, unless explicitly disabled with `config=False`. It is not directly apparent what `html-details-no-syntax-highlighting = yes` would mean. Actually, I think there's a bug in the config file handling because `config-key = no` would reset to the default, which would make the above option impossible to disable. :thinking:

Reply:
This is a Gold point. I haven't seen the existing negative Option. I also Luke the autogenerared - - no-... I'll Chance this in a separate PR.